File: 4-6.cpp
#include "iostream" 
#include "opencv2/opencv.hpp"   

using namespace std;
using namespace cv;

int main() {
    
    Mat img_color = imread("gray.jpeg", IMREAD_COLOR);

    if(img_color.empty()) {
        cout << "can nato read image file" << endl;
        return -1;
    }

    Mat img_gray;
    cvtColor(img_color, img_gray, COLOR_BGR2GRAY);

    Mat img_binary;
    threshold(img_gray, img_binary, 127, 255, THRESH_BINARY);

    imshow("GrayScale", img_gray);
    imshow("binary-threshold", img_binary);

    waitKey(0);
    destroyAllWindows();

    return 0;
}

File: 4-6.py
import cv2 as cv
import sys


img_color = cv.imread('gray.jpeg', cv.IMREAD_COLOR)

if img_color is None:
    print('can not read img_color')
    sys.exit(1)

img_gray = cv.cvtColor(img_color, cv.COLOR_BGR2GRAY)

retval,img_binary = cv.threshold(img_gray, 127 ,255, cv.THRESH_BINARY)
print(retval)
cv.imshow('GrayScale', img_gray)
cv.imshow('Binary', img_binary)
cv.waitKey(0)
cv.destroyAllWindows()
